### PR TITLE
fix: add confirmation modal when disabling WooPayments

### PR DESCRIPTION
**Pull request by frosso**
#### Changes proposed in this Pull Request

Discussed here: p1708097977644219-slack-CGGCLBN58

Showing a confirmation modal when the merchant decides to disable WooPayments.
This is the same dialog displayed in the main WC settings page.

#### Testing instructions

- Go to the WooPayments settings page http://wcpay.test/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Click on the checkbox to enable/disable WooPayments
- When disabling, a confirmation modal is displayed. Confirming on the modal will disable WooPayments.
- When enabling, no confirmation modal is displayed

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8222` or branch name `fix/show-confirmation-modal-on-disable` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/show-confirmation-modal-on-disable) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: bde022ddf311e024f48826edbfddc2a98af05f70
- Build time: 2024-02-19 09:42:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/disable-confirmation-modal/test/disable-confirmation-modal.test.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import userEvent from '@testing-library/user-event';
 /**
  * Internal dependencies
  */
-import DisableConfirmationModal from '../disable-confirmation-modal';
+import DisableConfirmationModal from '..';
```
Comment (frosso) [inline client/disable-confirmation-modal/test/disable-confirmation-modal.test.js:12]: I moved location of this component to a more "generalized" location, so obviously some references needed to be updated.

File: client/settings/general-settings/index.js [hunk 2]
```diff
@@ -9,50 +9,24 @@ import interpolateComponents from '@automattic/interpolate-components';
...
-			action: enableWCPay ? 'enable' : 'disable',
-			context: 'wcpay-settings',
-		} );
-	};
-
 	return (
 		<>
 			<Card>
 				<CardBody>
-					<CheckboxControl
```
Comment (frosso) [inline client/settings/general-settings/index.js:39]: I encapsulated this checkbox (and the confirmation dialog) in a new component, to make it easier to test.

File: client/settings/general-settings/test/general-settings.test.js [hunk 3]
```diff
@@ -50,27 +57,49 @@ describe( 'GeneralSettings', () => {
 		}
 	);
 
-	it.each( [ [ true ], [ false ] ] )(
```
Comment (frosso) [inline client/settings/general-settings/test/general-settings.test.js:53]: A confirmation modal is displayed, so I needed to refactor this test a bit. With the refactor I made, having the `it.each` wouldn't have made things easier to read.

**Review by mdmoore — APPROVED** [review]
Works as described. Nice find and fix! 🚢

**Pull request merged**